pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -169,9 +173,26 @@ def ints_to_pytimedelta(const int64_t[:] arr, box=False):
             result[i] = <object>NaT
         else:
             if box:
-                result[i] = Timedelta(value)
-            else:
+                result[i] = _timedelta_from_value_and_reso(value, reso=reso)
+            elif reso == NPY_DATETIMEUNIT.NPY_FR_ns:
                 result[i] = timedelta(microseconds=int(value) / 1000)
+            elif reso == NPY_DATETIMEUNIT.NPY_FR_us:
+                result[i] = timedelta(microseconds=value)
+            elif reso == NPY_DATETIMEUNIT.NPY_FR_ms:
+                result[i] = timedelta(milliseconds=value)
+            elif reso == NPY_DATETIMEUNIT.NPY_FR_s:
+                result[i] = timedelta(seconds=value)
+            elif reso == NPY_DATETIMEUNIT.NPY_FR_m:
+                result[i] = timedelta(minutes=value)
+            elif reso == NPY_DATETIMEUNIT.NPY_FR_h:
+                result[i] = timedelta(hours=value)
+            elif reso == NPY_DATETIMEUNIT.NPY_FR_D:
+                result[i] = timedelta(days=value)
+            elif reso == NPY_DATETIMEUNIT.NPY_FR_W:
+                result[i] = timedelta(weeks=value)
+            else:
+                # Month, Year, NPY_FR_GENERIC, pico, fempto, atto
+                raise NotImplementedError(reso)
```

Comment:
Is there tests for this error (if it's user facing)?

Reply:
no, will add one